service_configuration_lib/spark_config.py

```diff
@@ -1105,20 +1112,22 @@ def get_spark_conf(
         # Pick a port from a pre-defined port range, which will then be used by our Jupyter
         # server metric aggregator API. The aggregator API collects Prometheus metrics from multiple
         # Spark sessions and exposes them through a single endpoint.
-        try:
-            ui_port = int(
-                (spark_opts_from_env or {}).get('spark.ui.port') or
-                utils.ephemeral_port_reserve_range(
-                    self.spark_constants.get('preferred_spark_ui_port_start'),
-                    self.spark_constants.get('preferred_spark_ui_port_end'),
-                ),
-            )
-        except Exception as e:
-            log.warning(
-                f'Could not get an available port using srv-config port range: {e}. '
-                'Using default port range to get an available port.',
-            )
-            ui_port = utils.ephemeral_port_reserve_range()
+        ui_port: int = self.spark_constants.get('preferred_spark_ui_port_start', EPHEMERAL_PORT_START)
```

Comment:
could probably use a comment here that for tron-launched drivers, the ui port will essentially be hardcoded (i'm assuming that we're ignoring `spark.ui.port` on purpose, so this comment would let future-us know that we didn't forget about that option)

Reply:
It seems we are still relying on `spark.ui.port` for tron-launched spark drivers (iirc the hard-coded one was removed a while ago), and the tron "executor" pod's ports are retrieved from [get_spark_ports_from_config](https://sourcegraph.com/github.com/Yelp/paasta@84cb7ecd911dc24fcadc5cd901c2430b163c4243/-/blob/paasta_tools/tron_tools.py?L1053-1059) if `executor: spark`.

Reply:
anyway I added some more comments for make it more clear